core/src/main/java/hudson/Functions.java

```diff
@@ -1939,12 +1938,14 @@ public String getServerName() {
         String url = Jenkins.get().getRootUrl();
         try {
             if (url != null) {
-                String host = new URL(url).getHost();
+                String host = new URI(url).toURL().getHost();
                 if (host != null)
                     return host;
             }
         } catch (MalformedURLException e) {
             // fall back to HTTP request
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
```

Comment:
should this also ignore the exception line `MalformedURLException` does?

Reply:
@timja, in this case error is wrapped in `RuntimeException` only because the result should invariably be a valid URL. So, addition of a checked exception type that would do nothing more than make the code more verbose as it would never be thrown. Therefore, it is opted to throw a `RuntimeException` instead.

Reply:
but the `MalformedURLException` falls back to other code? i.e. this block is expected to be forgiving on invalid urls, e.g. empty strings.

Reply:
@timja please suggest if any on how to check that without breaking the code.

Reply:
@timja the block of code mentioned above is already marked `@Deprecated`. I made changes only related in resolution of [JENKINS-72028] issue.

Reply:
I see, I couldn't find any usages of it so should be safe enough